### PR TITLE
Move Round to util.cpp and optimize it.

### PR DESCRIPTION
**Pull request by denravonska**
Move Round to util.cpp and optimize it.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -8904,7 +8894,8 @@ int64_t ComputeResearchAccrual(int64_t nTime, std::string cpid, std::string oper
     double ReferencePPD = dMagnitudeUnit*dAvgMag;
     if ((PPD > ReferencePPD*5))
     {
-            printf("Researcher PPD %f > Reference PPD %f for CPID %s with Lifetime Avg Mag of %f, Days %f \r\n",PPD,ReferencePPD,cpid.c_str(),dAvgMag,days);
+            printf("Researcher PPD %f > Reference PPD %f for CPID %s with Lifetime Avg Mag of %f, Days %f, MagUnit %f\r\n",
+                   PPD,ReferencePPD,cpid.c_str(),dAvgMag,days, dMagnitudeUnit);
```
Comment (tomasbrod) [inline src/main.cpp]: you could also drop the \\r\\n while at it :)

File: src/util.cpp [hunk 2]
```diff
@@ -1446,11 +1448,20 @@ string FormatFullVersion()
 
 #endif
 
+double Round(double d, int place)
+{
+    const double accuracy = place != 0
+            ? std::pow(10, place)
+            : 1;
```
Comment (tomasbrod) [inline src/util.cpp]: >  pow(base, ±0) returns 1 for any base, even when base is NaN 

So no need for that check for base being zero. Interesting things happen when negative is passed.
Some may say, it rounds 6.022 to -1 places as expected to 10. If that is not desired, i propose the following modification.

```
double accuracy = place > 0
            ? std::pow(10, place)
            : 1;
```
Btw: avoid negation in condition of ternary operator.

File: src/util.cpp [hunk 3]
```diff
@@ -1446,11 +1448,20 @@ string FormatFullVersion()
...
+
+    return std::round(d * accuracy) / accuracy;
+}
+
 std::string RoundToString(double d, int place)
 {
     std::ostringstream ss;
-    ss << std::fixed << std::setprecision(place) << d ;
-    return ss.str() ;
+    ss << std::fixed << std::setprecision(place) << d;
```
Comment (tomasbrod) [inline src/util.cpp:1460]: This one is confusing, but after reading 4 manual pages on iostram and iomanipulator I think it does what is expected.
To this point I am not sure whether locale affects the decimal separator.

**Review by tomasbrod — CHANGES_REQUESTED** [review]

File: src/main.cpp (hunk 1 above)
Comment (denravonska) [reply src/main.cpp]: Will do. That one wasn't even supposed to be here but she can get a free ticket.

File: src/util.cpp (hunk 2 above)
Comment (denravonska) [reply src/util.cpp]: Oh I didn't know that. I think that 6.0 rounded to -1 decimals = 10 is exactly what we want. In that case, can we just shorten it to: `const double accuracy = std::pow(10, place);`?

Comment (tomasbrod) [reply src/util.cpp]: yes

File: src/util.cpp (hunk 3 above)
Comment (tomasbrod) [reply src/util.cpp:1460]: Aha, this is just a whitespace change.
Once changing [the above](#discussion_r132849848), I approve.

**Pull request merged**